src/core/Board.ts

```diff
@@ -71,6 +73,31 @@ class Board {
     BoardInitializer.init(this);
   }
 
+  findKing(color: Color): King {
+    const pieces = this.pieces;
+
+    for(const piece of pieces) {
+        if(piece.color == color && piece.type == PieceType.King) {
+            return piece;
+        }
+    }
+
+    throw new Error("Unexpected situation -> king with given color not found!");
+  }
+
+  getPiecesForColor(color: Color): Piece[] {
+    const pieces = this.pieces;
+    const colorPieces: Piece[] = []
+
+    for(const piece of pieces) {
+        if(piece.color === color) {
+            colorPieces.push(piece);
+        }
+    }
+
+    return colorPieces;
```

Comment:
This can be done in a much more elegant way using Array.filter method:
`return this.pieces.filter(piece => piece.color === color)`